py/server/deephaven/stream/kafka/producer.py

```diff
@@ -56,6 +59,8 @@ def produce(
         last_by_key_columns (bool): whether to publish only the last record for each unique key, Ignored if key_spec is
             KeyValueSpec.IGNORE. Otherwise, if last_by_key_columns is true this method will internally perform a last_by
             aggregation on table grouped by the input columns of key_spec and publish to Kafka from the result.
+        publish_initial (bool): whether the initial data in table should be published. When False, table.is_refreshing
+            must be True. By default, is True.
```

Comment:
Add a check for the case of  `publish_initial = False and not table.is_refreshing`?